Optimize SoftmaxOp on CPU

xiaomengy opened the pull request:
Summary: Optimize SoftmaxOp on CPU

Differential Revision: D14689516

BIT-silence commented:
Benchmark:

input size = [100, 100, 100, 100], axis = -1 on CPU: 370ms -> 300ms

xiaomengy committed:
Optimize SoftmaxOp on CPU (#18635)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/18635

Optimize SoftmaxOp on CPU

Differential Revision: D14689516

fbshipit-source-id: eef45c5762e15df6d15e643042a0be1fd20f71c5

houseroad reviewed (approved):
Looks good!
